File: Solution_Practice_I/java/prob2_3.java
import java.util.Scanner;
public class prob2_3 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int id;
        double purchase;
        do {
            System.out.println("Ingrese el ID y cantidad de compra");
            id=input.nextInt();
            purchase=input.nextInt();
        } while (id<=0 || purchase<0);
        if(purchase>1200){
            purchase-=100;
        }
        if(id<100){
            purchase-=(purchase*0.2);
        }else if(id<1000 && purchase>400){
            purchase-=(purchase*0.1);
        }
        System.out.printf("%.1f", purchase);
        input.close();
    }
}

File: Solution_Practice_I/python/prob2_3.py
while True:
    print("Ingrese el ID y cantidad de compra → (id cantidad)")
    id, purchase = map(float, input().split())
    int(id)
    if(purchase>=0 and id>0):
        break
if purchase>1200:
    purchase-=100

if id<100:
    purchase-=(purchase*0.2)
elif id<1000 and purchase>400:
    purchase-=(purchase*0.1)
print(f"{purchase:.1f}")
